lib/spack/spack/build_systems/cmake.py

```diff
@@ -94,6 +94,14 @@ class CMakePackage(PackageBase):
             description='CMake build type',
             values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
 
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_INTERPROCEDURAL_OPTIMIZATION.html
+    variant('ipo', default=False,
+            description='CMake interprocedural optimization')
+    # CMAKE_INTERPROCEDURAL_OPTIMIZATION only exists for CMake >= 3.9
+    conflicts('+ipo', when='^cmake@:3.8',
+              msg='CMAKE_INTERPROCEDURAL_OPTIMIZATION cannot be used with '
+                  'CMake < 3.9')
```

Comment:
Minor suggestion to shorten the message:
```suggestion
              msg='+ipo is not supported by CMake < 3.9')
```